### PR TITLE
Add iceCandidatePool option in trickle-ice example

### PR DESCRIPTION
**Pull request by juandebravo**
**Description**
Expose the option to configure iceCandidatePool in trickle-ice examples.

**Purpose**

Chrome 59 implements the iceCandidatePoolSize member of RTCConfiguration, that instructs the RTCPeerConnection instance to gather, as a performance optimization, ICE candidates before calling setLocalDescription. No news about when Firefox will implement this functionality.

http://www.juandebravo.com/2017/04/01/improve-setup-time-webrtc

screen shot 2017-04-13 at 14 37 55

### CODE REVIEW

File: src/content/peerconnection/trickle-ice/index.html [hunk 1]
```diff
@@ -82,12 +82,13 @@
...
-        <input type="radio" name="transports" value="all" id="all" checked><label for="all">all</label>
+        <label for="ipv6"><span>IceTransports value:</span></label>
+        <input type="radio" name="transports" value="all" id="all" checked>
+        <span>all</span>
         <input type="radio" name="transports" value="relay" id="relay">
-        <label for="relay">relay</label>
+        <span>relay</span>
         <input type="radio" name="transports" value="none" id="none">
-        <label for="none">none</label>
+        <span>none</span>
```
Comment (fippo) [inline src/content/peerconnection/trickle-ice/index.html]: none has actually been removed from the spec, can you remove it as a drive-by? :-)

File: src/content/peerconnection/trickle-ice/index.html [hunk 2]
```diff
@@ -97,6 +98,13 @@
         <label for="unmux">Gather RTCP candidates:</label>
         <input id="unmux" type="checkbox" checked>
       </div>
+      <div>
+        <label>ICE Candidate Pool:</label>
+        <span id="poolValue">0</span>
+        <span class="gray">0</span>
+        <input id="iceCandidatePool" type="range" min="0" max="10" value="0">
```
Comment (fippo) [inline src/content/peerconnection/trickle-ice/index.html:103]: i'd set a max of 1 here since we only have a single component.

File: src/content/peerconnection/trickle-ice/js/main.js [hunk 3]
```diff
@@ -128,29 +140,26 @@ function noDescription(error) {
 function parseCandidate(text) {
   var candidateStr = 'candidate:';
   var pos = text.indexOf(candidateStr) + candidateStr.length;
-  var fields = text.substr(pos).split(' ');
+  var [foundation, component, protocol, priority, address, port, , type] = text.substr(pos).split(' ');
```
Comment (fippo) [inline src/content/peerconnection/trickle-ice/js/main.js]: isn't this ES6? Should be supported in browsers we care about so i think this is ok.

File: src/content/peerconnection/trickle-ice/js/main.js [hunk 4]
```diff
@@ -128,29 +140,26 @@ function noDescription(error) {
...
 // ex: 126 | 32252 | 255 (126 is host preference, 255 is component ID 1)
 function formatPriority(priority) {
-  var s = '';
-  s += (priority >> 24);
-  s += ' | ';
-  s += (priority >> 8) & 0xFFFF;
-  s += ' | ';
-  s += priority & 0xFF;
-  return s;
+  var s = [priority >> 24,
```
Comment (fippo) [inline src/content/peerconnection/trickle-ice/js/main.js]: nit: return [...].join(' | ');

**Review by fippo — APPROVED** [review]
LGTM with nits. Thanks!

File: src/content/peerconnection/trickle-ice/js/main.js (hunk 3 above)
Comment (fippo) [reply src/content/peerconnection/trickle-ice/js/main.js]: maybe we should wait until we have https://github.com/webrtc/adapter/pull/527 ;-)

**Comment by KaptenJansson** [comment]
Thanks for the PR, could you fix the lint errors pls ?

**Comment by juandebravo** [comment]
@KaptenJansson lint errors fixed!

**Comment by KaptenJansson** [comment]
LGTM

**Comment by fippo** [comment]
\o/

btw: the next major version of adapter will add a shim that lets us get rid of parseCandidate
